maptemplate.c

```diff
@@ -1310,6 +1316,15 @@ static int processItemTag(layerObj *layer, char **line, shapeObj *shape)
       tagValue = msReplaceSubstring(tagValue, "$value", itemValue);
       msFree(itemValue);
 
+      if (padding != -1) {
+          int paddedSize = sizeof(tagValue) + (padding * sizeof(char*));
+          char *paddedValue = NULL;
+          paddedValue = (char *) msSmallMalloc(paddedSize + 1);
+          snprintf(paddedValue, paddedSize, paddingFormatString, padding, tagValue);
+          tagValue = msStrdup(paddedValue);
```

Comment:
you lack a msFree(tagValue) before.
And you could then just do tagValue = addedValue; to avoid the extra dance of msStrdup() / msFree()

Reply:
The following:
```
          tagValue = msStrdup(paddedValue);
          msFree(paddedValue);
```
should become?
```
          msFree(tagValue);
          tagValue = paddedValue;
```

Reply:
yes

Reply:
Updated